### PR TITLE
Fix URL Normalization code - fixes #13858

### LINKED ISSUE #13858: 'www.' URL normalization code is not correct
there are a bunch of places in the code that call something like `host = host.replace('www.', '')`, presumably for the purpose of considering 'www.example.com' and 'example.com' to be equal. 

this doesn't work on URLs like `wowww.com` (which turns into `wocom`).

### PR DESCRIPTION
**Pull request by catilac**
Resolves #13858

## Submitter Checklist:

- [X] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [X] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [X] Added/updated tests for this change (for new code or code which already has tests).
- [X] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).
- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review). (Ask a Brave employee to help if you cannot access this document.)

## Reviewer Checklist:

- [ ] Request a security/privacy review [as needed](https://github.com/brave/handbook/blob/master/development/security.md#how-to-request-a-security-review) if one was not already requested.

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

**Comment by ryanml** [comment]
@catilac thanks for your contribution :D 

For completions sake, can you add some tests for variations of `wwwow.com`? Let me know if you have any questions.

**Comment by catilac** [comment]
@ryanml thanks for your feedback! i can do that :)

File: test/unit/app/common/lib/suggestionTest.js [hunk 1]
```diff
@@ -58,6 +58,11 @@ describe('suggestion unit tests', function () {
     it('normalizes location', function () {
       assert.ok(suggestion.normalizeLocation('https://www.site.com') === 'site.com', 'www. prefix removed')
       assert.ok(suggestion.normalizeLocation('http://site.com') === 'site.com', 'location not modified')
+      assert.ok(suggestion.normalizeLocation('http://wowww.com') === 'wowww.com', 'location not modified')
```
Comment (diracdeltas) [inline test/unit/app/common/lib/suggestionTest.js]: these tests should not all say 'location not modified'

please also add one for `www.wowww.com`, thanks!

Comment (catilac) [reply test/unit/app/common/lib/suggestionTest.js]: oh, good catch! thanks. i'll make the changes now.

**Comment by diracdeltas** [comment]
master / 0.24.x: 68a51aa31462e1dc51a8e35be7951e7f51abcb82 (sorry this was not a merge commit - had to squash the commits into one)

**Comment by catilac** [comment]
@diracdeltas oops sorry about that

**Pull request merged**